ql/math/interpolations/cubicinterpolation.hpp

```diff
@@ -575,7 +575,7 @@ namespace QuantLib {
                                 for (Size i=1; i<n_-1; ++i) {
                                     Real Smin = std::min(S_[i-1], S_[i]);
                                     Real Smax = std::max(S_[i-1], S_[i]);
-                                    tmp_[i] = 3.0*Smin*Smax/(Smax+2.0*Smin);
+                                    tmp_[i] = Smin == 0 || Smax == 0 ? 0 : 3.0*Smin*Smax/(Smax+2.0*Smin);
```

Comment:
This is where the division by zero occur, but I'm not sure of the condition.  This catches the case in which the denominator is null because both `Smin` and `Smax` are null, but doesn't check if `Smax+2.0*Smin` is null for some other reason; see the dataset `x=[0,1,2,3,4]`, `y=[2,1,0,2,3]` which still fails.  I'm not sure what we should do in that case, though.